panther_analysis_tool/log_schemas/user_defined.py

```diff
@@ -0,0 +1,279 @@
+                result.api_response = response
+            results.append(result)
+        return results
+
+    @staticmethod
+    def _load_from_yaml(files: List[str]) -> Dict[str, ProcessedFile]:
+        yaml_parser = YAML(typ="safe")
+
+        processed_files = {}
+        for filename in files:
+            logger.info('Processing schema in file %s', filename)
+            processed_file = ProcessedFile()
+            processed_files[filename] = processed_file
+            try:
+                with open(filename, "r") as schema_file:
+                    processed_file.raw = schema_file.read()
+                processed_file.yaml = yaml_parser.load(processed_file.raw)
+            except (ParserError, ScannerError) as exc:
+                processed_file.error = f"invalid YAML: {exc}"
+        return processed_files
+
+    def _extract_schema_name(
+            self,
+            definition: Optional[Dict[str, Any]]
+    ) -> Tuple[str, Optional[str]]:
+        if definition is None:
+            raise ValueError('definition cannot be None')
+
+        name = definition.get('schema')
+
+        if name is None:
+            return "", "key 'schema' not found"
+
+        if not name.startswith(self._SCHEMA_NAME_PREFIX):
+            return "", f"'schema' field: value must start" \
+                       f" with the prefix '{self._SCHEMA_NAME_PREFIX}'"
+
+        return name, None
+
+    def _update_or_create_schema(
+            self,
+            name: str,
+            processed_file: ProcessedFile
+    ) -> Tuple[bool, bool, Dict[str, Any]]:
+        existing_schema = self.find_schema(name)
+        current_reference_url = ''
+        current_description = ''
+        current_revision = 0
+        definition = cast(Dict[str, Any], processed_file.yaml)
+        existed = False
+        if existing_schema is not None:
+            existing_schema = cast(Dict[str, Any], existing_schema)
+            existed = True
+            current_reference_url = existing_schema.get('referenceURL', '')
+            current_description = existing_schema.get('description', '')
+            current_revision = existing_schema['revision']
+        reference_url = definition.get('referenceURL', current_reference_url)
+        description = definition.get('description', current_description)
+        logger.debug('updating schema %s at revision %d, using '
+                     'referenceURL=%s, '
+                     'description=%s',
+                     name,
+                     current_revision,
+                     reference_url,
+                     description)
+        success, response = self.api_client.put_schema(
+            name=name,
+            definition=processed_file.raw,
+            revision=current_revision,
+            reference_url=reference_url,
+            description=description,
+        )
+        return existed, success, response
+
+
+def discover_files(base_path: str, pattern: str = '*.yml') -> List[str]:
```

Comment:
I would also say include `*.yaml` in there. We support `*.yaml` in other tools (e.g. `pantherlog`)

Reply:
Sure, will do.